rpc/endpoints.go

```diff
@@ -22,11 +22,37 @@ import (
 	"github.com/ethereum/go-ethereum/log"
 )
 
+// mustAvailableModule enforces that requested api modules (eg. via --rpcapi) are actually
+// available API services. If an invalid module is given (ie API "foo" wanted which does not exist),
+// then log.Crit is used to cause program to exit, logging the invalid module and a list of available
+// API service names.
+func mustAvailableModule(module string, apis []API) {
+	for _, api := range apis {
+		if module == api.Namespace {
+			return
+		}
+	}
+	log.Crit("invalid api module", "module", module, "available", func() (available []string) {
```

Comment:
Please change the code to return an error instead of calling `log.Crit`.

Reply:
3ac3773